<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1771</width>
    <height>888</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>195</width>
      <height>528</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <spacer name="verticalSpacer_10">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="Choose_Camera">
         <property name="text">
          <string>Choose Cam</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="0">
        <widget class="QFrame" name="frame">
         <property name="minimumSize">
          <size>
           <width>110</width>
           <height>200</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <widget class="QWidget" name="layoutWidget">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>96</width>
            <height>185</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Contrast</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="Contrast">
             <property name="maximum">
              <number>200</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
             <property name="sliderPosition">
              <number>100</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_6">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>18</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Brightness</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="Brightness">
             <property name="minimum">
              <number>-50</number>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_7">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>18</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="Grayscale">
             <property name="text">
              <string>Grayscale</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item row="7" column="0">
        <spacer name="verticalSpacer_8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Filters</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QComboBox" name="Edge_detection">
           <item>
            <property name="text">
             <string>Choose</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Sobel</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Canny</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Prewitt</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Laplacian</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="18" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="19" column="0">
        <widget class="QPushButton" name="Save_File">
         <property name="text">
          <string>Save File</string>
         </property>
        </widget>
       </item>
       <item row="20" column="0">
        <spacer name="verticalSpacer_9">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="13" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Min</string>
         </property>
        </widget>
       </item>
       <item row="13" column="1">
        <widget class="QSpinBox" name="Canny_Min">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="14" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Max</string>
         </property>
        </widget>
       </item>
       <item row="14" column="1">
        <widget class="QSpinBox" name="Canny_Max">
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="value">
          <number>200</number>
         </property>
        </widget>
       </item>
       <item row="15" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Scale</string>
         </property>
        </widget>
       </item>
       <item row="16" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Delta</string>
         </property>
        </widget>
       </item>
       <item row="17" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Kernel Size</string>
         </property>
        </widget>
       </item>
       <item row="17" column="1">
        <widget class="QSpinBox" name="Kernel_Spinner">
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>21</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
         <property name="value">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item row="16" column="1">
        <widget class="QDoubleSpinBox" name="Delta_Spinner"/>
       </item>
       <item row="15" column="1">
        <widget class="QDoubleSpinBox" name="Scale_Spinner">
         <property name="minimum">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="Open_File">
         <property name="text">
          <string>Open File</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="Output_image">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>150</y>
      <width>611</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="Input_label">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>80</y>
      <width>59</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Input Image</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Output_label">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>80</y>
      <width>67</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Output Image</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QGraphicsView" name="Input_image">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>150</y>
      <width>611</width>
      <height>391</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>384</width>
      <height>216</height>
     </size>
    </property>
   </widget>
   <widget class="QCameraViewfinder" name="viewfinder" native="true">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>150</y>
      <width>611</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCameraViewfinder</class>
   <extends>QWidget</extends>
   <header>qcameraviewfinder.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
